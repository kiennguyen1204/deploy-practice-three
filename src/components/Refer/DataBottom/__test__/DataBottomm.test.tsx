import { render, screen } from '@testing-library/react';

// components
import ReferFriendItem from 'components/Refer/DataBottom';

describe('ReferFriendItem Component', () => {
  it('renders the component with the correct title', () => {
    const props = {
      index: 1,
      icon: <img src='icon.png' alt='Icon' />,
      title: 'Sample Title',
      description: 'Sample Description'
    };

    const { asFragment } = render(<ReferFriendItem {...props} />);

    // Check if the title  are rendered
    const titleElement = screen.getByText(props.title);

    expect(titleElement).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
