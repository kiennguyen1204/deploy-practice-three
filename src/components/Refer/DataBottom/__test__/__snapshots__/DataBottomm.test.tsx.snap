// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferFriendItem Component renders the component with the correct title 1`] = `
<DocumentFragment>
  <article
    class="item-bottom"
  >
    <section
      class="flex-start-container top-header"
    >
      <div
        class="flex-container index-circle"
      >
        2
      </div>
      <div
        class="icon"
      >
        <img
          alt="Icon"
          src="icon.png"
        />
      </div>
    </section>
    <section
      class="item-content"
    >
      <h2
        class="title"
      >
        Sample Title
      </h2>
      <p
        class="description"
      >
        Sample Description
      </p>
    </section>
  </article>
</DocumentFragment>
`;
